ccm_web/client/src/pages/AddUMUsers.tsx

```diff
@@ -0,0 +1,450 @@
+                <Button variant="outlined" >Submit</Button>
+              </Paper>
+            </Grid>
+          </Box>
+        </Grid>
+        <Backdrop className={classes.backdrop} open={isAddingEnrollments}>
+        <Grid container>
+          <Grid item xs={12}>
+            <CircularProgress color="inherit" />
+          </Grid>
+          <Grid item xs={12}>
+          Loading...
+          </Grid>
+        </Grid>
+      </Backdrop>
+      </div>)
+  }
+
+  const setStateUpload = (): void => {
+    setEnrollments(undefined)
+    setErrors(undefined)
+    setActiveStep(States.UploadCSV)
+  }
+
+  const renderUploadAgainButton = (): JSX.Element => {
+    return <Button color='primary' component="span" onClick={() => setStateUpload()}>Upload again</Button>
+  }
+
+  const renderErrors = (errors: ValidationError[]): JSX.Element => {
+    return (
+      <div>
+        {renderCSVFileName()}
+        <Grid container justify='flex-start'>
+          <Box clone order={{ xs: 2, sm: 1 }}>
+            <Grid item xs={12} sm={9} className={classes.parseErrorTable} >
+              <ValidationErrorTable invalidations={errors} />
+            </Grid>
+          </Box>
+          <Box clone order={{ xs: 1, sm: 2 }}>
+            <Grid item xs={12} sm={3} className={classes.parseErrorDialog}>
+              <Paper role='alert'>
+                <Typography>Review  your CSV file</Typography>
+                <ErrorIcon className={classes.parseErrorIcon} fontSize='large'/>
+                <Typography>Correct the file and{renderUploadAgainButton()}</Typography>
+              </Paper>
+            </Grid>
+          </Box>
+        </Grid>
+      </div>)
+  }
+
+  const getShouldNotHappenContent = (): JSX.Element => {
+    return (<div>Unexpected step</div>)
+  }
+
+  const getStepContent = (stepIndex: number): JSX.Element => {
+    switch (stepIndex) {
+      case 0:
+        return getSelectContent()
+      case 1:
+        return getUploadContent()
+      case 2:
+        return getReviewContent()
+      default:
+        return getShouldNotHappenContent()
+    }
+  }
+
+  const handleReset = (): void => {
+    setActiveStep(States.SelectSection)
+  }
+
+  return (
+    <>
+      <div className={classes.root}>
+        <Typography variant='h5'>{addUMUsersProps.title}</Typography>
+        <Stepper activeStep={activeStep} alternativeLabel>
+          {steps.map((label) => (
+            <Step key={label}>
+              <StepLabel>{label}</StepLabel>
+            </Step>
+          ))}
+        </Stepper>
+        <div>
+          {activeStep === steps.length
+            ? (
+                <div>
+                  <Typography className={classes.instructions}>All steps completed</Typography>
+                  <Button onClick={handleReset}>Reset</Button>
+                </div>
+              )
+            : (
+                <div>
+                  {getStepContent(activeStep)}
```

Comment:
Not needed, Can you pass in the step label ( like Select, Upload, Review, Confirmation) instead  of the step number. I feel you could retrieve it but I may be wrong.

That way in the switch statement the case option could be ('Select', 'Upload', 'Review', 'Confirmation') instead of 0,1,2 making it readable code

Reply:
Yup, should have done this.  I added the enum later and forgot to update it. ty

Reply:
Actually not 100% sure I addressed what you were referring to based on the lines github is showing.  I did change the switch statement to use the enum by name, though.